extensions/functions_arithmetic.yaml

```diff
@@ -573,6 +573,63 @@ aggregate_functions:
         decomposable: MANY
         intermediate: fp64?
         return: fp64?
+  - name: "product"
+    description: Product of a set of values. Returns null for empty input.
```

Comment:
Having thought about this some more, let's just go with the sane version of product that returns 1 on empty input like @julianhyde is suggesting. If sum really is the only function that the SQL spec messed up, let's not use it as precedent to mess up other functions as well. That is, update the description, remove the `?` from the intermediate and return types, and remove the `nullability: DECLARED_OUTPUT` lines (`MIRROR` is the default mode).

Reply:
```suggestion
    description: Product of a set of values. Returns 1 for empty input.
```